### ISSUE TITLE
Subscribe Block showing HTML text formatting tags in CTA Button text in the live site

### ISSUE THREAD

**Issue by sudeepbaral** [Automattic/jetpack#25104]
### Quick summary

CTA Button in Subscribe Block displays HTML Text Formatting tags in the live site when the texts are italicized or bold but  not on in the editor. This happens across many themes.

![Markup on 2022-07-11 at 1:00:32 PM](https://user-images.githubusercontent.com/3326057/178209137-2376ebad-7fb1-4259-bb13-aa883cd073fe.png)

### Steps to reproduce

1. Edit a Page/Post
2. Add Subscribe Block
3. Change the text in the CTA button to Italic or bold
4. Update and Preview the post/page

### What you expected to happen

The text in the CTA button should have been italicized or bold without the HTML tags <em> </em> and <strong> </strong> being visible in the live site and preview.

### What actually happened

The texts in CTA button are showing with  the HTML tags <em> </em> and <strong> </strong> added in the live site and preview.

### Context

User reported in 5362876-zen and shared they did not see the HTML tags before in the live site.

- [ ] 5362876-zen

### Simple, Atomic or both?

Simple, Atomic

### Theme-specific issue?

This has happened across all themes I tested : Dicot, Twenty Twenty Two, Twenty Twenty One, Twenty Twenty, Marl, Twenty Sixteen

### Reproducibility

Consistent

### Severity

Some (< 50%)

**Comment by Robertght** [comment]
📌 HOUSEKEEPING
- Labels ✅ 
- Priority ✅ 
- Replicable on Core - Yes/No

📌 SCRUBBING
- Tested on P2 - no
- Tested on Simple ✅ 
- Tested on AT ✅ 
- Tested on Self-hosted ✅ 

📌 FINDINGS/SCREENSHOTS/VIDEO
- Also occurs on self-hosted sites. I'm moving this to the Jetpack repo cc @jeherve for further visibility

📌 ACTIONS
- Marked as Triaged for Quality Squad review
- Transferred to XYZ external repo: [link](https://github.com/Automattic/jetpack/)

**Comment by jeherve** [comment]
@wojtekn Do you think you could take a look? I think it may have been caused by #24763.

**Comment by wojtekn** [comment]
@jeherve yes, it is caused by https://github.com/Automattic/jetpack/pull/24763.

I've checked it a bit, and I'm unsure how to approach it. Technically it is more correct than it was - internally, we save Subscribers block as a shortcode, submit text is a shortocode attribute, and the shortcode attribute should have HTML escaped per [Shortcode API docs](https://codex.wordpress.org/Shortcode_API#Attributes). However, it breaks `em`, `strong`, and `s` tags, which we can use for submit button text in the editor.

We could revert to the previous behavior by removing escaping for "<" and ">" characters which would make those HTML tags work again. The downside is that it breaks when we try to save the button with the ">" character.

Any thoughts? Do you know other blocks that use shortcode attributes to store HTML?

**Comment by jeherve** [comment]
> Do you know other blocks that use shortcode attributes to store HTML?

This is the only one, as we've tried to move away from shortcodes in our blocks. The Subscribe block will eventually have the same fate: #15227

> We could revert to the previous behavior by removing escaping for "<" and ">" characters which would make those HTML tags work again. The downside is that it breaks when we try to save the button with the ">" character.

Would there be a way to only remove escaping for the exact tags we support, and nothing else?
https://github.com/Automattic/jetpack/blob/b0a09443dd803ee4817811d94a4c66cc3c4261dd/projects/plugins/jetpack/modules/subscriptions/views.php#L17-L27

**Comment by wojtekn** [comment]
@jeherve it looks that subscribe block deserves special treatment for now :-) 

I checked it more, and I think it may be better to leave the current encoding as is - that way we are producing shortcode code that follows the documentation, and we store potentially unsafe data with HTML entities encoded.

I would try to solve it at display time then - to unescape those four tags there. I've investigated it more and as far as I see, `wp_kses` function which is used to display text should handle our case OOTB. The problem is that now it gets encoded `$subscribe_button` value. If we decode HTML entities before passing button text to `wp_kses`, it seems working fine:


![Screen Shot 2022-07-15 at 18 31 18](https://user-images.githubusercontent.com/727413/179271297-095bad48-d5a8-4e1c-8235-8c6761aa3f09.png)

I will run more tests, and if it works well, I will prepare PR next week.
